### PR TITLE
gitlab: 13.7.4 -> 13.8.4

### PR DESCRIPTION
**Pull request by ghost**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Ran manual tests:

- create a repository
- edit a file via the web ui

### CODE REVIEW

**Comment by petabyteboy** [comment]
CC @jslight90

File: pkgs/development/ruby-modules/gem-config/default.nix [hunk 1]
```diff
@@ -201,14 +201,35 @@ in
...
   gitlab-markup = attrs: { meta.priority = 1; };
 
-  gitlab-pg_query = attrs: lib.optionalAttrs (attrs.version == "1.3.0") {
+  gitlab-pg_query = attrs: if (attrs.version == "1.3.0") then {
     dontBuild = false;
     postPatch = ''
       sed -i 's;"https://gitlab.com.*";"${fetchurl {
         url = "https://gitlab.com/gitlab-org/libpg_query/-/archive/gitlab-10-1.0.3/libpg_query-gitlab-10-1.0.3.tar.gz";
-        sha256 = "1519x4v6wrk189mjg4hlfah0f7hjy3syg8kk8b6g644gdspzs26j";
+        sha256 = "0000000000000000000000000000000000000000000000000000";
```
Comment (ajs124) [inline pkgs/development/ruby-modules/gem-config/default.nix:209]: This… looks weird? Does this just not get evaluated?

Comment (petabyteboy) [reply pkgs/development/ruby-modules/gem-config/default.nix:209]: Oh, right! This isn't evaluated for GitLab 13.8.4. Should we leave it in (and revert to the correct hash) or remove support for building gitlab-pg_query 1.3.0 from gem-config entirely?

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112836` at 00b133e4 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>gitlab</li>
    <li>gitlab-ee</li>
  </ul>
</details>
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/53b7ecffd57221fe2b4f4341ea10f1a1">gitaly</a></li></li>
    <li><a href="https://gist.github.com/23b68af533f5d7bfe529f6d03354941d">gitlab-shell</a></li></li>
    <li><a href="https://gist.github.com/420026fda8ae821192f8033ebe22224b">gitlab-workhorse</a></li></li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>error: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-flags-array.md">no-flags-array</a>

`buildFlagsArray` is only intended to be used in Bash, not as a Nix attribute.

Near pkgs/applications/version-management/gitlab/gitlab-workhorse/default.nix:17:3:
```
   |
17 |   buildFlagsArray = "-ldflags=-X main.Version=${version}";
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/version-management/gitlab/gitaly/default.nix:30:23:
```
   |
30 |           patches = [ ../fix-grpc-ar.patch ];
   |                       ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/version-management/gitlab/gitlab-shell/default.nix:15:15:
```
   |
15 |   patches = [ ./remove-hardcoded-locations.patch ];
   |               ^
```
</li>
  </ul>
</details>

File: pkgs/development/ruby-modules/gem-config/default.nix (hunk 1 above)
Comment (ajs124) [reply pkgs/development/ruby-modules/gem-config/default.nix:209]: I'm kind of wondering why this is checking for the version in the first place. If it's needed, it should be applied and if not, it should be dropped IMO. Or can there be multiple versions of `pg_query` in this one gemset?

**Comment by marsam** [comment]
I think these changes should be backported to stable https://about.gitlab.com/releases/2021/02/11/security-release-gitlab-13-8-4-released/

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 112836` at 00b133e4 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages failed to build:</summary>
  <ul>
    <li>gitaly</li>
    <li>gitlab</li>
    <li>gitlab-ee</li>
    <li>gitlab-shell</li>
    <li>gitlab-workhorse</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>error: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-flags-array.md">no-flags-array</a>

`buildFlagsArray` is only intended to be used in Bash, not as a Nix attribute.

Near pkgs/applications/version-management/gitlab/gitlab-workhorse/default.nix:17:3:
```
   |
17 |   buildFlagsArray = "-ldflags=-X main.Version=${version}";
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/version-management/gitlab/gitaly/default.nix:30:23:
```
   |
30 |           patches = [ ../fix-grpc-ar.patch ];
   |                       ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-patch-comment.md">missing-patch-comment</a>

Please add a comment on the line above, explaining the purpose of this patch.
Near pkgs/applications/version-management/gitlab/gitlab-shell/default.nix:15:15:
```
   |
15 |   patches = [ ./remove-hardcoded-locations.patch ];
   |               ^
```
</li>
  </ul>
</details>

File: pkgs/development/ruby-modules/gem-config/default.nix (hunk 1 above)
Comment (petabyteboy) [reply pkgs/development/ruby-modules/gem-config/default.nix:209]: Not sure about one gemset, but this is the global nixpkgs gem-config, so another gemset could use the old version still.

Comment (petabyteboy) [reply pkgs/development/ruby-modules/gem-config/default.nix]: I verified that no other gemsets in nixpkgs are using the old version of gitlab-pg_query using nix-review.

**Pull request merged**